sample/github/README.md

```diff
@@ -6,156 +6,130 @@ deployed to receive the webhook's event deliveries and forward them into a
-`spec.trigger.resource` in `samples/github/flow.yaml`.  
-
-And then apply the resources:
-
-```shell
-ko apply -f sample/github/auth.yaml
-ko apply -f sample/github/flow.yaml
-```   
-
-Once deployed, you can inspect the created resources with `kubectl` commands:
-
-```shell
-# This will show the available Flows we created:
-kubectl get flows -o yaml
-
-# This will show the available Feeds created from the Flow:
-kubectl get feeds -o yaml
-
-# This will show the available Channels created from the Flow:
-kubectl get channels -o yaml
+## Granting permissions
 
-# This will show the available Subscriptions created from the Flow:
-kubectl get subscriptions -o yaml
+Because the `github` EventSource needs to create a `Service.serving.knative.dev`
+and call the Kubernetes watch API on events, you'll need to provision a special
+ServiceAccount with the necessary permissions.
 
-# This will show the available ClusterBuses we created in the prerequisites:
-kubectl get clusterbuses -o yaml
-```
-
-Then create the secret so that you can see changes:
+The `eventing/samples/github-events/auth.yaml` file provisions a service account, creates a role
+which can create Service in the `default` namespace and can view all
+Kubernetes resources. In a production environment, you might want to limit the
+access of this service account to only specific namespaces.
 
 ```shell
- ko apply -f sample/github/githubsecret.yaml
+kubectl apply -f eventing/samples/github-events/auth.yaml
 ```
 
-Then create a PR for the repo you configured the webhook for, and you'll see that the Title
-will be modified with the suffix '(looks pretty legit)'
+## Build and deploy the sample
+
+1.  Use Docker to build the sample code into a container. To build and push with
+    Docker Hub, run these commands replacing `{username}` with your Docker Hub
+    username. Run the following from the _root_ of the `knative/docs` repo:
+
+    ```shell
+    # Build the container on your local machine
+    docker build -t {username}/github-events --file=eventing/samples/github-events/Dockerfile .
+
+    # Push the container to docker registry
+    docker push {username}/github-events
+    ```
+
+1.  After the build has completed and the container is pushed to Docker Hub, you
+    can deploy the function into your cluster. **Ensure that the container image
+    value in `function.yaml` matches the container you built in the previous
+    step.** Apply the configuration using `kubectl`:
+
+    ```shell
+    kubectl apply -f eventing/samples/github-events/function.yaml
+    ```
+
+1.  Check that your service is running using:
+
+    ```shell
+    kubectl get services.serving.knative.dev -o "custom-columns=NAME:.metadata.name,READY:.status.conditions[2].status,REASON:.status.conditions[2].message"
+    NAME              READY     REASON
+    legit             True      <none>
+    ```
+
+1.  Create a [personal access token](https://github.com/settings/tokens) to 
```

Comment:
I'd also call out that we use the same token for both the flow (to set up the webhook) and the function (to read and update the PR title).

I'd also move this step up before step 2 (`kubectl apply -f function.yaml`), so that the secrets are in place and the service has everything it needs before step 3 (wait for service to become ready).